Bio/Align/psl.py

```diff
@@ -0,0 +1,486 @@
+# Copyright 2021 by Michiel de Hoon.  All rights reserved.
+#
+# This file is part of the Biopython distribution and governed by your
+# choice of the "Biopython License Agreement" or the "BSD 3-Clause License".
+# Please see the LICENSE file that should have been included as part of this
+# package.
+"""Bio.Align support for the "psl" multiple alignment format.
+
+The Pattern Space Layout (PSL foramt, described by UCSC, stores a series
+of pairwise alignments in a single file. Typically they are used for
+transcript to genome alignments. PSL files stored the alignment positions
+and alignment scores, but do not store the aligned sequences.
+
+See http://genome.ucsc.edu/FAQ/FAQformat.html#format5
+
+You are expected to use this module via the Bio.Align functions.
+
+Coordinates in the PSL format are defined in terms of zero-based start
+positions (like Python) and aligning region sizes.
+
+A minimal aligned region of length one and starting at first position in the
+source sequence would have ``start == 0`` and ``size == 1``.
+
+As we can see on this example, ``start + size`` will give one more than the
+zero-based end position. We can therefore manipulate ``start`` and
+``start + size`` as python list slice boundaries.
+"""
+from itertools import chain
+import numpy
+
+
+from Bio.Align import Alignment
+from Bio.Align import interfaces
+from Bio.Seq import Seq, reverse_complement, UndefinedSequenceError
+from Bio.SeqRecord import SeqRecord
+
+
+class AlignmentWriter(interfaces.AlignmentWriter):
+    """Alignment file writer for the Pattern Space Layout (PSL) file format."""
+
+    def __init__(self, target, mode="w", header=True, mask=None, wildcard="N"):
+        """Create an AlignmentWriter object.
+
+        Arguments:
+         - target    - output stream or file name
+         - header    = If True (default), write the PSL header consisting of
```

Comment:
Switch the = to a - to match the rest, but does this render nicely as RST?

Reply:
fixed